sentence_transformers/cross_encoder/losses/ListMLELoss.py

```diff
@@ -0,0 +1,290 @@
+            )
+
+    def forward(self, inputs: list[list[str], list[list[str]]], labels: list[Tensor]) -> Tensor:
+        """
+        Compute ListMLE loss for a batch of queries and their documents.
+
+        Args:
+            inputs: List of (queries, documents_list)
+            labels: Ground truth relevance scores, shape (batch_size, num_documents)
+
+        Returns:
+            Tensor: Mean ListMLE loss over the batch
+        """
+        if isinstance(labels, Tensor):
+            raise ValueError(
+                "ListMLELoss expects a list of labels for each sample, but got a single value for each sample."
+            )
+
+        if len(inputs) != 2:
+            raise ValueError(
+                f"ListMLELoss expects two inputs (queries, documents_list), but got {len(inputs)} inputs."
+            )
+
+        queries, docs_list = inputs
+        docs_per_query = [len(docs) for docs in docs_list]
+        max_docs = max(docs_per_query)
+        batch_size = len(queries)
+
+        if docs_per_query != [len(labels) for labels in labels]:
+            raise ValueError(
+                f"Number of documents per query in inputs ({docs_per_query}) does not match number of labels per query ({[len(labels) for labels in labels]})."
+            )
+
+        pairs = [(query, document) for query, docs in zip(queries, docs_list) for document in docs]
+
+        if not pairs:
+            # Handle edge case where there are no documents
+            return torch.tensor(0.0, device=self.model.device, requires_grad=True)
+
+        mini_batch_size = self.mini_batch_size or batch_size
+        if mini_batch_size <= 0:
+            mini_batch_size = len(pairs)
+
+        logits_list = []
+        for i in range(0, len(pairs), mini_batch_size):
+            mini_batch_pairs = pairs[i : i + mini_batch_size]
+
+            tokens = self.model.tokenizer(
+                mini_batch_pairs,
+                padding=True,
+                truncation=True,
+                return_tensors="pt",
+            )
+            tokens = tokens.to(self.model.device)
+
+            logits = self.model(**tokens)[0].view(-1)
+            logits_list.append(logits)
+
+        logits = torch.cat(logits_list, dim=0)
+        logits = self.activation_fct(logits)
+
+        # Create output tensor filled with a very small value for padded logits
+        logits_matrix = torch.full((batch_size, max_docs), -1e16, device=self.model.device)
+
+        # Place logits in the desired positions in the logit matrix
+        doc_indices = torch.cat([torch.arange(len(docs)) for docs in docs_list], dim=0)
+        batch_indices = torch.repeat_interleave(torch.arange(batch_size), torch.tensor(docs_per_query))
+        logits_matrix[batch_indices, doc_indices] = logits
+
+        # Create a mask for valid entries
+        mask = torch.zeros((batch_size, max_docs), dtype=torch.bool, device=self.model.device)
```

Comment:
```suggestion
        mask = torch.zeros_like(logits_matrix, dtype=torch.bool)
```
This is a bit simpler, also makes it obvious that it's a mask over the logits matrix.